proposal.rst

```diff
@@ -268,7 +285,29 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+
+* Weeks of bonding: Get more familiar with ranking algorithms, select specific suitable parts of the above datasets, discuss what is wrong with `#278 <https://github.com/xapian/xapian/pull/278>`_.
+
+* 1 Week - FIRE dataset integration. Each user have to download it by themself, because of 
+registration required. Implement training process using Letor API.  
+
+* 2 Weeks of exams (situation with exams is very unstable now in my University, so I timeline may change in this case).
+
+* 1 Week - add function for selecting scorer and ranker. It would be nice to write API allows user to select default sets of queries to run and run specified queries. For that purpose update build system to be able to run suite from terminal.
+
+* 1 Week - TREC dataset integration. It has similar format to FIRE, it should not be problem to integrate it. Start writing tool for combine docs.
+
+* 1 Week - combine tool API should support plain logical operations, it could take time to implement them.
+
+* 1 Week - prepare tool for text data extracting from Wikipedia.
+
+* 1 Week - add functionallity to perftest to drop system cache for more accurate results and run one case multiple times to get average time consuming.
+
+* 1 Week - implement Wikipedia data loader and queries for it, write performance tests for filling db with documents, iterating over documents.
+
+* 1 Week - add perftest to autotools srcipt, as it advised in `#107 <https://trac.xapian.org/ticket/107>`_. If time is left, add function to fork test case in new process and measure it memory usage.
```

Comment:
`perftest` is already integrated in the autotools build system - that isn't what #107 is about at all.

The fundamental problem with `perftest` is it is entirely based on randomly-generated data, and that means it isn't representative of real-world use (or at least we can't trust that it is).  That makes profiling and optimisation based on `perftest` results dubious at best.

We want a performance test suite based on real world data, or at least data that's very carefully randomly generated to match real data in ways that matter (but just using real data seems simpler).